### PR TITLE
Add additional documentation to application spec

### PR DESCRIPTION
**Pull request by sreya**

### CODE REVIEW

File: workspaces/applications.md [hunk 1]
```diff
@@ -20,46 +20,68 @@ UI:
 
 ## Application specification file
 
-The application specification file allows you to define what Coder needs to
+In order to specify custom applications, add a configuration file at path
```
Comment (khorne3) [inline workspaces/applications.md:23]:
```suggestion
To define custom applications, add a configuration file at
```

File: workspaces/applications.md [hunk 2]
```diff
@@ -20,46 +20,68 @@ UI:
...
+        path: "/healthz"
 ```
 
 **Notes**:
 
-- All top-level fields in the `config.yaml` file are required
-- You must include at least one health check. The `exec` health check looks for
-  an exit code of `0`, while the `http` health check looks for the return of
-  `HTTP 200`.
+- A health check _must_ report healthy in order to access the application.
```
Comment (khorne3) [inline workspaces/applications.md:83]:
```suggestion
- A health check _must_ report healthy for you to access the application.
```

File: workspaces/applications.md [hunk 3]
```diff
@@ -20,46 +20,68 @@ UI:
...
+- If both HTTP and Exec health checks are specified then HTTP is preferred.
```
Comment (khorne3) [inline workspaces/applications.md:84]:
```suggestion
- If you specify both the HTTP and Exec health checks, Coder prioritizes HTTP.
```

**Review by khorne3 — APPROVED** [review]

**Review by bpmct — APPROVED** [review]
Looks great!

**Pull request merged**